In [2]:
#Import the required Libraries.
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

## Data Cleaning 

### Data Reading & Data Types 

In [5]:
#Read the data in pandas
inp0= pd.read_csv("Attribute+DataSet.csv")
inp1= pd.read_csv("Dress+Sales.csv")

You have “Attribute DataSet” which contains a column named “Price”. Choose the correct statement from the following about its data type and variable type.
- Integer type and numerical variable
- Object type and categorical ordinal variable
- Object type and categorical nominal variable
- Float type and categorical variable.


There is another column in “Attribute DataSet” named as “Recommendation”, choose the correct statement about its data type and variable type.
- Integer type and categorical
- Object type and categorical
- Integer type and continuous numerical
- Object type only.


Which of the following column do you think are of no use in “Attribute DataSet”.
- Dress_ID
- Price
- Size and material
- NeckLine
- None of the above


In [10]:
# Print the information about the attributes of inp0 and inp1.
inp0.Style.value_counts()

Casual      219
Sexy         66
party        51
cute         44
vintage      25
bohemian     23
Brief        17
work         17
sexy          7
Novelty       6
Flare         2
fashion       1
OL            1
Name: Style, dtype: int64

### Fixing the Rows and Columns 

As you can see, there is a column in “Attribute Dataset” named as ‘Size’. This column contains the values in abbreviation format. Write a code in Python to convert the followings:

- M into  “Medium”
- L into  “Large”
- XL into “Extra large”
- free into “Free”
- S, s & small into “Small”.

Now once you are done with changes in the dataset, what is the value of the lowest percentage, the highest percentage and the percentage of Small size categories in the column named “Size”?


In [ ]:
# Column fixing, correcting size abbreviation. count the percentage of each size category in "Size" column.


In [ ]:
# Print the value counts of each category in "Size" column.


### Impute/Remove Missing values

In [ ]:
# Print the null count of each variables of inp0 and inp1.


You are given another dataset named “Dress Sales”. Now if you observe the datatypes of the columns using ‘inp1.info()’ command, you can identify that there are certain columns defined as object data type though they primarily consist of numeric data.

Now if you try and convert these object data type columns into numeric data type(float), you will come across an error message. Try to correct this error.







In [ ]:
# Print the data types information of inp1 i.e. "Dress Sales" data.


In [ ]:
# Try to convert the object type into float type of data. YOU GET ERROR MESSAGE.


In [ ]:
# Do the required changes in the "Dress Sales" data set to get null values on string values.


In [ ]:
# Convert the object type columns in "Dress Sales" into float type of data type.


When you see the null counts in “Dress Sales” dataset after performing all the operations that have been mentioned in jupyter notebook, you will find that there are some columns in “Dress Sales” data where there are more than 40% of missing values. Based on your understanding of dealing with missing values do the following steps.

In [ ]:
# Print the null percetange of each column of inp1.


In [ ]:
# Drop the columns in "Dress Sales" which have more than 40% of missing values.


You should categorise the dates into seasons in “Dress Sales” data to simplify the analysis according to the following criteria:
- June, July and August: Summer.
- September, October and November: Autumn.
- December, January and February: WInter.
- March, April and May: Spring.




In [ ]:
# Create the four seasons columns in inp1, according to the above criteria.


In [ ]:
# calculate the sum of sales in each seasons in inp1 i.e. "Dress Sales".


Now let's merge inp1 with inp0 with left join manner, so that the information of inp0 should remain intact.

In [19]:
# Merge inp0 with inp1 into inp0. this is also called left merge.
inp0 = pd.merge(left=inp0,right=inp1, how='left', left_on='Dress_ID', right_on='Dress_ID')
inp0.head()
inp0.Style.value_counts(normalize=True)

Casual      0.457203
Sexy        0.137787
party       0.106472
cute        0.091858
vintage     0.052192
bohemian    0.048017
Brief       0.035491
work        0.035491
sexy        0.014614
Novelty     0.012526
Flare       0.004175
fashion     0.002088
OL          0.002088
Name: Style, dtype: float64

In [13]:
# Now Drop the Date columns from inp0 as it is already combined into four seasons.
inp0.drop(inp0.loc[:,'29-08-2013':'10-12-2013'].columns, axis= 1, inplace= True)
inp0.isnull().sum()

KeyError: '29-08-2013'

Print the null count of inp0 to get the idea about the missing values in data set.

In [20]:
# Print the null count of each columns in inp0 dataframe i.e. combined data frame of inp0 and inp1 without date columns.
inp0

Dress_ID    Style    Price  Rating  Size  Season   NeckLine  \
0    1006032852     Sexy      Low     4.6     M  Summer     o-neck   
1    1212192089   Casual      Low     0.0     L  Summer     o-neck   
2    1190380701  vintage     High     0.0     L  Automn     o-neck   
3     966005983    Brief  Average     4.6     L  Spring     o-neck   
4     876339541     cute      Low     4.5     M  Summer     o-neck   
..          ...      ...      ...     ...   ...     ...        ...   
474   990559192    Brief  Average     4.7     M  winter     o-neck   
475   713391965   Casual      Low     4.7     M  Spring     o-neck   
476   532874347   Casual  Average     4.7     M  Summer     v-neck   
477   655464934   Casual  Average     4.6     L  winter  boat-neck   
478   919930954   Casual      Low     4.4  free  Summer     v-neck   

    SleeveLength       Material  FabricType  ... 24-09-2013_y 26-09-2013_y  \
0      sleevless            NaN     chiffon  ...         3554       3624.0   
1          Petal     microfiber         NaN  ...         2710       2942.0   
2           full       polyster         NaN  ...           11         11.0   
3           full           silk     chiffon  ...         1878       1892.0   
4      butterfly  chiffonfabric     chiffon  ...         2032       2156.0   
..           ...            ...         ...  ...          ...          ...   
474   halfsleeve        acrylic     chiffon  ...           73         74.0   
475         full       polyster         NaN  ...          400        388.0   
476         full         cotton         NaN  ...          616        597.0   
477    sleevless           silk  broadcloth  ...          257        256.0   
478        short         cotton    Corduroy  ...          628        632.0   

     28-09-2013_y  30-09-2013_y  10-02-2013_y  10-04-2013_y  10-06-2013_y  \
0            3706        3746.0        3795.0        3832.0          3897   
1            3258        3354.0        3475.0        3654.0          3911   
2              11          11.0          11.0          11.0            11   
3            1914        1924.0        1929.0        1941.0          1952   
4            2252        2312.0        2387.0        2459.0          2544   
..            ...           ...           ...           ...           ...   
474            75          75.0          76.0          76.0            77   
475           360         364.0         372.0         377.0           380   
476           586         569.0         561.0         555.0           551   
477           255         254.0         253.0         250.0           249   
478           639         645.0         651.0         655.0           660   

     10-08-2013_y  10-10-2013_y  10-12-2013_y  
0          3923.0        3985.0          4048  
1          4024.0        4125.0          4277  
2            11.0          11.0            11  
3          1955.0        1959.0          1963  
4          2614.0        2693.0          2736  
..            ...           ...           ...  
474          77.0          77.0            77  
475         382.0         384.0           285  
476         546.0         535.0           520  
477         249.0         249.0           248  
478         668.0         674.0           680  

[479 rows x 197 columns]

You can see that there are two types of variables one with a large number of missing values and another is very less number of missing values. These two columns can be categorized as:

Type-1: Missing values are very less (around 2 or 3 missing values): Price, Season, NeckLine, SleeveLength, Winter and Autumn. 

Type-2: Missing values are large in numbers (more than 15%): Material, FabricType, Decoration and Pattern Type.



In [ ]:
# Deal with the missing values of Type-1 columns: Price, Season, NeckLine, SleeveLength, Winter and Autumn.


In [ ]:
# Deal with the missing values for Type-2 columns: Material, FabricType, Decoration and Pattern Type.


### Standardise value 

In the given dataset, there are certain discrepancies with the categorical names such as irregular spellings. Choose the correct option of columns with irregular categories and update them.
 
- Season, NeckLine
- Price, Material
- fabricType, Decoration
- Season, SleeveLength


In [ ]:
#correcting the spellings.


In [ ]:
#correcting the Spellings.


## Univariate Analysis 

### Categorical Unordered Univariate Analysis
 

There is a column named ‘Style’ in ‘Attribute Dataset’ which consists of the different style categories of the women apparels. Certain categories whose total sale is less than 50000 across all the seasons is considered under one single category as ‘Others’.


Which of the following categories in ‘Style’ column can be grouped into ‘Others’ category? and perform the grouping operation in the notebook for further analysis.
- Flare, fashion
- Novelty, bohemian
- OL, fashion, work
- Novelty, fashion, Flare


In [2]:
# Group "Style" categories into "Others" which have less than 50000 sales across all the seasons.


What is the percentage of “cute” and “Others” category in “Style” column in “Attribute DataSet” respectively?
- 46%, 5%
- 9%, 2.1%
- 2.1%, 5%
- 13.8%, 9%


In [ ]:
# Calculate the percentage of each categories in the "Style" variable.


Similarly Club Neckline, SLeeve length categories into "Others" which have less than 50000 sales across all the seasons.

In [ ]:
# Group "Neckline" categories into "Others" which have less than 50000 sales across all the seasons.


In [ ]:
# Group "Sleeve length" categories into "Others" which have less than 50000 sales across all the seasons.


Club material, fabrictype, patterntype and decoration categories into "Others" which have less than 25000 sales across all the seasons

In [ ]:
# Group "material" categories into "Others" which have less than 25000 sales across all the seasons.


In [ ]:
# Group "fabric type" categories into "Others" which have less than 25000 sales across all the seasons.


In [ ]:
# Group "patern type" categories into "Others" which have less than 25000 sales across all the seasons.


In [ ]:
# Group "decoration" categories into "Others" which have less than 25000 sales across all the seasons.


### Caregorical Ordered Univariate Analysis

Which of the following is an unordered variable in “Attribute DataSet”.
- Style
- Price
- Season
- Size


### Numerical variable Univariate analysis:

What is the approximate difference between the maximum value and 75th percentile in “Autumn” column.
- Approx 54000
- Approx 55000
- Approx 52000
- Approx 50000



In [ ]:
# Describe the numerical variale: "Autumn".


In [ ]:
# plot the boxplot of "Autumn" column.


Which of the following season has the highest difference between the maximum value and 99th quantile of sales?
- Winter
- Summer
- Spring
- Autumn


In [ ]:
# Find the maximum and 99th percentile of Winter season.


In [ ]:
# Find the maximum and 99th percentile of Summer season.


In [ ]:
# Find the maximum and 99th percentile of Spring season.


In [ ]:
# Find the maximum and 99th percentile of Autumn season.


## Bivariate Analysis 

### Numerical- Categorical analysis

Which of the following “Price” category has the lowest average value of rating?
- very-high
- Medium
- Low
- High


In [ ]:
# Find the Mean of Ratings for each Price category.


What is the median of the rating of “vintage” category in Style column?
- 4.6
- 4.7
- 4.55
- 0.00


In [ ]:
# Find the median of Ratings for each Style category.


Which of the following season has the highest average value of sale for “Recommendation” value equals to 1.
- Summer
- Spring
- Autumn
- Winter


In [ ]:
# Summer sale vs Recommendation.


In [ ]:
# Spring sale vs Recommendation.


In [ ]:
# Autumn sale vs Recommendation.


In [ ]:
# Winter sale vs Recommendation.


### Categorical categorical bivariate analysis


Which of the following size categories has the highest positive recommendations?
- Medium and extra large
- Extra large and small
- Free and small
- Free and medium


In [ ]:
# Size vs Recommendation.


### Multivariate analysis 

Which of the following pair of “Style” and “Price” category has the highest average of positive recommendations?
- Price: medium and style: vintage
- Price: medium and style: cute
- Price: very high and style: party
- Price: low and style: sexy


In [ ]:
# plot the heat map of Style, price and Recommendation.


Which of the following material type has no recommendation in summer and winter seasons?
- Mix and Milksilk
- Nylon and Rayon
- Microfiber and Silk
- Milksilk and Microfiber


In [ ]:
# plot the heat map of Season, material and Recommendation.
